qip-0009.md

```diff
@@ -0,0 +1,529 @@
+2:    currentHeader ← startHeader  // Initialize currentHeader with startHeader
+3:    prevHeader ← startHeader  // Initialize prevHeader with startHeader
+4:    currentRank ← CalculateRank(currentHeader)  // Calculate the rank of the currentHeader
+5:    Initialize followDownHeaders as an empty array of headers
+
+6:    while true do
+7:        if currentHeader.Number > endHeader.Number then
+8:            if currentRank > 0 then
+9:                currentRank ← currentRank - 1
+10:           else
+11:               currentRank ← 0
+12:           end if
+13:       else
+14:           // Only update prevHeader if we are allowed to go backwards
+15:           prevHeader ← currentHeader
+16:           followDownHeaders ← append(followDownHeaders, currentHeader)
+17:       end if
+18:       
+19:       // Get the header referenced in the interlink of prevHeader at currentRank
+20:       currentHeader ← GetHeader(prevHeader.Interlink(currentRank))
+21:       
+22:       if currentHeader.Hash == endHeader.Hash then
+23:           break
+24:       end if
+25:   end while
+26:   
+27:   return followDownHeaders
+28: end function
+
+```
+
+```
+1: function followUpForward(Header, txHash, endOrder)
+2:    Initialize infixHeaders as an empty array of headers
+3:    order ← Order(Block)  // Get the order of the current block
+4:    infixHeaders ← append(infixHeaders, Block)  // Add the current block to infixHeaders
+5:    children ← Header  // Initialize children with Header
+6:    var lastHeader as Header
+
+7:    if order <= endOrder then
+8:        while true do
+9:            if order == endOrder then
+10:               lastHeader = children
+11:               break
+12:           end if
+13:           
+14:           headerNumber ← children.Number(order)
+15:           children ← GetHeader(headerNumber + 1)
+16:           childrenOrder = Order(children)
+17:           
+18:           // Move up a level if a dominant header is found
+19:           if childrenOrder > order then
+20:               order = childrenOrder
+21:           end if
+22:           infixHeaders = append(infixHeaders, children)
+23:       end while
+24:   else
+25:       lastHeader = Header
+26:   end if
+27:   
+28:   return (infixHeaders, lastHeader)
+29: end function
+
+```
+
+
+## Verifier Execution
+
+Verifier is a node/blockchain interface sdk that does not have a complete knowledge of the blockchain and has to verify and pick the best proof from the honest party between a pool of Proofs provided by the Honest parties or the Adversary. Adversary is a node that generates invalid proofs.
+
+Let $P$ be the pool of proofs and verifier has to pick the honest party proof to verify the Transaction execution in a Block. PickBestProof algorithm below describes the process. It does three things, 
+ * Checks if the proof is valid
+ * Check that the proof has continuous chain of blocks
+ * Compares the total intrinsic bits of entropy from all the proofs and picks the best proof
```

Comment:
Can clarify that it picks the one with the most entropy removed.

Reply:
yeah, thats a better description